Applications/Utils/FileConverter/MeshToRaster.cpp

```diff
@@ -0,0 +1,205 @@
+                                               : mesh->getMinEdgeLength();
+    INFO ("Cellsize set to %f", cellsize);
+
+    std::size_t const nNodes (mesh->getNumberOfNodes());
+    std::vector<MeshLib::Node*> const& nodes_vec(mesh->getNodes());
+    GeoLib::AABB bounding_box(nodes_vec.begin(), nodes_vec.end());
+    MathLib::Point3d const& min(bounding_box.getMinPoint());
+    MathLib::Point3d const& max(bounding_box.getMaxPoint());
+    std::size_t const n_cols = static_cast<std::size_t>(std::ceil((max[0] - min[0]) / cellsize));
+    std::size_t const n_rows = static_cast<std::size_t>(std::ceil((max[1] - min[1]) / cellsize));
+
+    // raster header
+    std::ofstream out(output_arg.getValue(), std::ios::out);
+    out << "ncols         " << n_cols+1 << "\n";
+    out << "nrows         " << n_rows+1 << "\n";
+    out << std::fixed << "xllcorner     " << (min[0] - cellsize / 2.0) << "\n";
+    out << std::fixed << "yllcorner     " << (min[1] - cellsize / 2.0) << "\n";
+    out << std::fixed << "cellsize      " << cellsize << "\n";
+    out << "NODATA_value  " << "-9999\n";
+    INFO("Writing raster with %d x %d pixels.", n_cols, n_rows);
+
+    MeshLib::MeshElementGrid const grid(*mesh);
+    double const max_edge (mesh->getMaxEdgeLength() + cellsize);
+
+    // pixel values
+    double x = min[0];
+    double y = max[1];
+    double const half_cell = cellsize/2.0;
+    for (std::size_t j=0; j<=n_rows; ++j)
+    {
+        for (std::size_t i=0; i<=n_cols; ++i)
+        {
+            MeshLib::Node const node(x, y, 0);
+            MathLib::Point3d min_vol{{x - max_edge, y - max_edge, -std::numeric_limits<double>::max()}};
+            MathLib::Point3d max_vol{{x + max_edge, y + max_edge, std::numeric_limits<double>::max()}};
+            std::vector<const MeshLib::Element*> const elems = grid.getElementsInVolume(min_vol, max_vol);
+            std::size_t const elem_idx = getProjectedElement(elems, node);
+            // centre of the pixel is located within a mesh element
+            if (elem_idx != std::numeric_limits<std::size_t>::max())
+                out << getElevation(*(mesh->getElement(elem_idx)), node) << " ";
+            // centre of the pixel isn't located within a mesh element
+            else
+            {
+                std::array<double, 4> const x_off{{-half_cell, half_cell, -half_cell, half_cell}};
+                std::array<double, 4> const y_off{{-half_cell, -half_cell, half_cell, half_cell}};
+                double sum (0);
+                std::size_t nonzero_count (0);
+                // test for all of the pixel's corner if any are within an element
+                for (std::size_t i=0; i<4; ++i)
+                {
+                    MeshLib::Node const node(x+x_off[i], y+y_off[i], 0);
+                    std::size_t const elem_idx = getProjectedElement(elems, node);
+                    if (elem_idx != std::numeric_limits<std::size_t>::max())
```

Comment:
This if condition seems to be always false, because it is in the else branch of the same test in line 169

Reply:
:heavy_check_mark: